grails-app/services/com/unifina/service/UserService.groovy

```diff
@@ -120,6 +122,17 @@ class UserService {
 		}
 	}
 
+	def delete(SecUser user, Long id) {
+		if (user == null || user.id == null || id == null) {
+			throw new NotFoundException("user not found", "User", null)
+		}
+		if (user.id != id) {
+			throw new ApiException(400, "DELETE_USER_ERROR", "delete user error")
```

Comment:
Hmm what about having message "can only delete own user account" or something similar. Allows user of API to more accurately diagnose what the cause was.